src/decode.h

```diff
@@ -351,6 +351,8 @@ typedef struct PktProfiling_ {
 /* forward declartion since Packet struct definition requires this */
 struct PacketQueue_;
 
+enum mode {IDS, IPS, AUTO};
```

Comment:
This is not addressed: https://github.com/inliniac/suricata/pull/1660#discussion_r41411237

Please make sure all comments are addressed.